glean-core/csharp/Glean/Glean.cs

```diff
@@ -343,6 +350,83 @@ internal void HandleBackgroundEvent()
             GleanInternalPings.events.Submit(EventsReasonCodes.background);
         }
 
+        /// <summary>
+        /// Collect and submit a ping for eventual upload.
+        /// 
+        /// The ping content is assembled as soon as possible, but upload is not
+        /// guaranteed to happen immediately, as that depends on the upload
+        /// policies.
+        /// 
+        /// If the ping currently contains no content, it will not be assembled and
+        /// queued for sending.
+        /// </summary>
+        /// <param name="ping">Ping to submit.</param>
+        /// <param name="reason">The reason the ping is being submitted.</param>
+        internal void SubmitPing(PingTypeBase ping, string reason = null)
+        {
+            SubmitPingByName(ping.name, reason);
+        }
+
+        /// <summary>
+        /// Collect and submit a ping for eventual upload by name.
+        /// 
+        /// The ping will be looked up in the known instances of `PingType`. If the
+        /// ping isn't known, an error is logged and the ping isn't queued for uploading.
+        /// 
+        /// The ping content is assembled as soon as possible, but upload is not
+        /// guaranteed to happen immediately, as that depends on the upload
+        /// policies.
+        /// 
+        /// If the ping currently contains no content, it will not be assembled and
+        /// queued for sending, unless explicitly specified otherwise in the registry
+        /// file.
+        /// </summary>
+        /// <param name="name">Name of the ping to submit.</param>
+        /// <param name="reason">The reason the ping is being submitted.</param>
+        internal void SubmitPingByName(string name, string reason = null)
+        {
+            Dispatchers.LaunchAPI(() => {
```

Comment:
Format nit, what does StyleCop think of this curly brace on the same line?

Reply:
Mh, it was, but really not that visible in the IDE. I'll file a bug to add linting to CI :)